index.js

```diff
@@ -20,30 +20,6 @@ const pluginOptions = {
 	markup: true
 };
 
-function makeHot(id, code, hotOptions) {
```

Comment:
I noticed that @rixo's branch does not have the below options specified in `pluginOptions`

```
	externalDependencies: true,
	hotReload: true,
	hotOptions: true,
	shared: true,
```

I think `shared` does not exist and perhaps we don't want to enable the others by default?

Reply:
Where did you get this from?

Reply:
The `svelte-hmr` branch: https://github.com/rixo/svelte-loader/tree/svelte-hmr

Reply:
I think we should only disable `hotReload` by default and remove `shared`. The other two doesn't do anything in `true/false` manner.

Reply:
It looks like the only thing it's used for is to determine if it's a compiler option: https://github.com/sveltejs/svelte-loader/blob/0f1c49b4e74734f7a76d5675dc2be96c6398ae49/index.js#L81

Since none of these are compiler options we should probably remove them all

Reply:
> It looks like the only thing it's used for is to determine if it's a compiler option:
> 
> https://github.com/sveltejs/svelte-loader/blob/0f1c49b4e74734f7a76d5675dc2be96c6398ae49/index.js#L81
> 
> Since none of these are compiler options we should probably remove them all

Doing that breaks tests and compilation.
This prevents passing described options to the compiler.

Reply:
I think it is better to avoid making changes that are out of the scope of this PR.
Code could be improved, but in the other PR.

Reply:
Just to clarify, when I said "all" I meant the four options I originally suggested removing: `externalDependencies`, `hotReload`, `hotOptions`, `shared`

There should be any reason to pass those to the compiler since none of those four are valid compiler options: https://svelte.dev/docs#svelte_compile

Reply:
> Just to clarify, when I said "all" I meant the four options I originally suggested removing: `externalDependencies`, `hotReload`, `hotOptions`, `shared`

It breaks tests and compilation. That options are **PREVENTED** from going to the compiler due to the `pluginOptions` object and code above.

Reply:
Oh, I see. Sorry I had that backwards. It's a bit confusingly written. In `rollup-plugin-svelte` we moved them to be under `compilerOptions`. We should probably do that here too, but that can be a separate change. Just removing `shared` and leaving the rest as-is would make sense

Reply:
It is done.

Yeah, that made me to scratch my head for some minutes too. :smile: